### PR TITLE
Use an ingress controller on the control-plane network for UI and API (through UI) access

### PR DESCRIPTION
**Pull request by NicolasT**
Cfr. https://github.com/scality/metalk8s/issues/1800#issuecomment-536082408

Still a bit of work to be done w.r.t. inter-service TLS traffic, and eventually turning this into a single *Ingress* object, but this should be a good start.

TODO:
- [x] Docs
- [ ] Consider using the host-network for the ingress controller, iff we can bind the resulting `nginx` to only the control-plane IP of a node, to recover source/client IPs (which doesn't work with `externalIPs` in a *Service*)

### CODE REVIEW

**Comment by NicolasT** [comment]
This is a bit stuck on the Cypress tests, because we now expose the UI on HTTPS using an IP-address and a self-signed certificate.

I believe this could be fixed by https://github.com/cypress-io/cypress/pull/4947 (cfr. https://github.com/cypress-io/cypress/issues/771), but there's no Cypress release including this change, and I seem to be unable to install an unreleased version for testing :roll_eyes:

**Comment by NicolasT** [comment]
Jay, found a work-around :tada:

File: docs/quickstart/services.rst [hunk 1]
```diff
@@ -64,23 +47,8 @@ This page displays two monitoring indicators:
...
-Forward the port used by Grafana:
-
-.. code-block:: shell
-
-   user@your-host $ kubectl -n metalk8s-monitoring port-forward svc/prometheus-operator-grafana 3000:80
-
-Then open your web browser and navigate to ``http://localhost:3000``
-
+Grafana is available on the same host as the MetalK8s UI, under ``/grafana``.
+Log in with the default credentials: ``grafana`` / ``prom-operator``.
```
Comment (sayf-eddine-scality) [inline docs/quickstart/services.rst]: Still having issue to login the default creds are not working

File: charts/nginx-ingress-control-plane.yaml [hunk 2]
```diff
@@ -0,0 +1,42 @@
+controller:
+  image:
+    repository: '{%- endraw -%}{{ build_image_name("nginx-ingress-controller", False) }}{%- raw -%}'
+
+  defaultBackendService: 'metalk8s-ingress/nginx-ingress-default-backend'
```
Comment (sayf-eddine-scality) [inline charts/nginx-ingress-control-plane.yaml:5]: we may need a good 404 page with the same theme as the UI

**Review by sayf-eddine-scality — CHANGES_REQUESTED** [review]
two minor comments the rest is LGTM

File: charts/nginx-ingress-control-plane.yaml (hunk 2 above)
Comment (gdemonet) [reply charts/nginx-ingress-control-plane.yaml:5]: Indeed, could even redirect to MetalK8s UI 404 page.

Comment (NicolasT) [reply charts/nginx-ingress-control-plane.yaml:5]: Good idea, but I think that's out of scope for this PR.

File: docs/quickstart/services.rst (hunk 1 above)
Comment (NicolasT) [reply docs/quickstart/services.rst]: That's because I made a mistake in the docs ;-P Will fix.

Comment (NicolasT) [reply docs/quickstart/services.rst]: I changed the default one to `admin`/`admin`. Should get OIDC up soon!

**Review by sayf-eddine-scality — APPROVED** [review]

**Pull request merged**